### PR TITLE
Fix release build on GCC 10 and add to CI

### LINKED ISSUE #1327: warning: writing 1 byte into a region of size 0 when compiling
~/s25client/build$ make
[  3%] Built target translations
[  3%] Built target glad
[  4%] Built target boost_nowide
[  5%] Built target endian_static
[  8%] Built target s25util_common
[ 16%] Built target siedler2
[ 17%] Built target s25util_log
[ 18%] Built target s25util_network
[ 19%] Built target mygettext
[ 20%] Built target s25Common
[ 21%] Built target rttrConfig
[ 22%] Built target SGE
[ 23%] Built target gamedata
[ 26%] Built target s25edit
[ 26%] Built target s25update
[ 27%] Built target samplerate
[ 27%] Built target audiodrv
[ 28%] Built target videodrv
[ 28%] Built target videoSDL2
[ 28%] Built target audioSDL
[ 28%] Built target drivers
[ 29%] Built target lobby
[ 29%] Built target lobby_c
[ 29%] Building CXX object libs/s25main/CMakeFiles/s25Main.dir/GamePlayer.cpp.o
/home/jens-ungerer/s25client/libs/s25main/GamePlayer.cpp: In member function ‘void GamePlayer::Deserialize(SerializedGameData&)’:
/home/jens-ungerer/s25client/libs/s25main/GamePlayer.cpp:328:22: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
  328 |         toolsSetting = sgd.PopUnsignedChar();
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/home/jens-ungerer/s25client/libs/s25main/GamePlayer.cpp:328:22: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
/home/jens-ungerer/s25client/libs/s25main/GamePlayer.cpp:328:22: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
/home/jens-ungerer/s25client/libs/s25main/GamePlayer.cpp:328:22: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
cc1plus: all warnings being treated as errors
make[2]: *** [libs/s25main/CMakeFiles/s25Main.dir/build.make:355: libs/s25main/CMakeFiles/s25Main.dir/GamePlayer.cpp.o] Fehler 1
make[1]: *** [CMakeFiles/Makefile2:1889: libs/s25main/CMakeFiles/s25Main.dir/all] Fehler 2
make: *** [Makefile:182: all] Fehler 2

### PR DESCRIPTION
**Pull request by Flamefire**
Fixes #1327

TODO: Move that code into a `popCollection` method (WIP but this PR should already improve things)

### CODE REVIEW

File: libs/s25main/GamePlayer.cpp [hunk 1]
```diff
@@ -324,8 +324,15 @@ void GamePlayer::Deserialize(SerializedGameData& sgd)
     for(uint8_t& militarySetting : militarySettings_)
         militarySetting = sgd.PopUnsignedChar();
 
+#if defined(__GNUC__) && __GNUC__ == 10
```
Comment (Flow86) [inline libs/s25main/GamePlayer.cpp:327]: make a comment to the issue link, so we can find it why we have added it

Comment (Flamefire) [reply libs/s25main/GamePlayer.cpp:328]: Done

**Comment by Flamefire** [comment]
I added also a removal of the -fno-strict-aliasing flag. I think this may have been required ages ago when we used UB to deserialize and fix endianess or so. We have `-Wstrict-aliasing=2` enabled to any violations should have been detected by now. Running a quick test shows no issues so far but with GCC-6. Working on that.

Quick benchmark using the integration test shows a runtime improvement of about 1/6 (0.14s less on ~0.7s total runtime)

**Comment by Flow86** [comment]
did you test the network protocol code too? an alignment could be problematic there?

**Comment by Flow86** [comment]
you need to skip the submodule update in this one (rebase please)

**Comment by Flamefire** [comment]
Waiting for https://github.com/Return-To-The-Roots/libsiedler2/pull/16 and will include the formatting fix forlibutil

**Comment by Flamefire** [comment]
Rebased fixed and added a bit refactoring involving the submodule to avoid having to update that again right after this.

**Pull request merged**